### DISCUSSION TITLE
High memory usage of my Vaultwarden instance on a 4VCPU 4GB VPS

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [dani-garcia/vaultwarden#4981] [Q&A]
Hey ! I noticed that the vaultwarden process is using 1GB in memory on my VPS, i'm using Docker with this image `vaultwarden/server:latest` and since I have 6000 passwords store the process is using 1GB and even more sometimes... Maybe a GC issue ?

Have a nice day !

**Comment by BlackDex** [comment]
Rust doesn't have GC, so that will not be an issue. 
Large vaults could have an impact. Maybe downloading favicons also. And maybe a database connection. 

You would need to check what is in memory to see what could be it.

You could also try the `alpine` images, those use a different memory allocator.
